### DISCUSSION TITLE
Rename commands imported from a module for testing and scripting

### DISCUSSION THREAD

**Discussion by volodymyrprokopyuk** [nushell/nushell#14637] [Q&A] [answered]
Hi,

Is is possible to rename commands imported from a module?

I have two files the `wallet.nu` and the `wallet-test.nu` in a directory.

The `wallet.nu` file defines and exports with the `main` prefix commands **directly accessible from the command line** through the `wallet.nu` script.  
```nushell
#!/usr/bin/env -S nu --stdin

export def "main address verify" []: string -> bool { ... }

def main [] { }

# usage: "address" | ./wallet address verify
```

The `wallet-test.nu` file imports the definitions from the `wallet.nu` for testing. Just because I want the `address verify` command to be directly exposed from the `wallet.nu` script, I'm forced to define and export the command with the `main` prefix. Is there a way to have both the exposed command in the script and renamed imports without the `main` prefix?
```nushell
#!/usr/bin/env nu

use std assert
use wallet.nu ["main address checksum" , "main address verify"] # I'd like to suppress `main` 

def "test address checksum" [] {
  assert equal ("address" | main address verify) true # I'd like to suppress `main`
}

test address checksum
```

**Comment by ghost** [comment, marked as answer]
i see 2 options here:

#### Option 1: Export the correct name from `wallet.nu`

(i would choose this since it also helps if anything else needs to import it)

```nu
export def "address verify" []: string -> bool {
  print "Hello"
  false
}

# no need for a `export`
# in fact exporting something prefixed with `main` can cause issues later
# (especially when nushell decides to change the handling of something related to it in a future update)
alias "main address verify" = address verify

def main [] {
  print -e "Missing argument"
}
```

#### Option 2: Create a alias within the `wallet-test.nu`

```nu
# import it as usual
use ./wallet.nu "main address verify"

# create a alias with the correct name
alias av = main address verify

# use it
"" | av
```

**Reply by volodymyrprokopyuk** [reply]
Yes, using aliases is an option that works. However, the above situation is quite recurring, and should have a more seamless integration into the language.
- I'd like to see exported commands directly accessible from the containing script/module without the `main` prefix
- I'd like import renaming to be directly supported by the `use` statement at the language level

Thank you!
